### PR TITLE
Refactor IG Documentation

### LINKED ISSUE #3309: docs: Update ig document
The `ig` document (https://www.inspektor-gadget.io/docs/latest/reference/ig) is outdated. Some changes it needs:
- [ ] Rework use cases making it clear it's not only useful when the API server is down
- [ ] Remove "Common features" as it's already explained in https://www.inspektor-gadget.io/docs/latest/reference/run
- [ ] Document TLS support 
- [ ] Move too specific things like "Adding ig in your own container image" to a different page?

### PR DESCRIPTION
**Pull request by andoriyaprashant**
## How to use

This pull request refactors the Inspektor Gadget (IG) documentation to enhance clarity and organization.

## Changes Made:

**Reworked Use Cases:**

Updated the "Use Cases" section to clarify that IG is valuable in scenarios beyond when the Kubernetes API server is down.

**Documented TLS Support:**

Added instructions for configuring client-side TLS support, including the --tls-ca-file flag for secure communication between the IG client and server.

**Removed "Common Features":**

The "Common Features" section has been removed, as these details are already covered in the [Run](https://www.inspektor-gadget.io/docs/latest/reference/run) documentation.

## Testing done

 [x] Rework use cases making it clear it's not only useful when the API server is down
 [x] Remove "Common features" as it's already explained in https://www.inspektor-gadget.io/docs/latest/reference/run
 [x] Document TLS support
 [ ] Move too specific things like "Adding ig in your own container image" to a different page

### CODE REVIEW

**Comment by andoriyaprashant** [comment]
Hello @mauriciovasquezbernal 
Could you please provide some guidance on this #3309 
Thank you

File: docs/reference/ig.md [hunk 1]
```diff
@@ -107,63 +105,34 @@ RUNTIME.RUNTIMENAME RUNTIME.CONTAINERID
...
-test-host                3326093    3326070    sh               0   /usr/bin/sh -c ls > /dev/null
-test-host                3326123    3326093    ls               0   /usr/bin/ls
+Example:
+
+```bash
+$ ig list-containers --tls-ca-file /path/to/ca.pem --remote-address https://ig-server-address
 ```
 
-Events generated from containers have their container field set, while events which are generated from the host do not.
+With these configurations, you can ensure that the communication between the ig client and server is encrypted and secure.
```
Comment (mauriciovasquezbernal) [inline docs/reference/ig.md:135]: ... between ig and gadgetctl

File: docs/reference/ig.md [hunk 2]
```diff
@@ -27,15 +27,13 @@ Some characteristics of `ig`:
...
 - If you don't want to install `kubectl-gadget` on your machine, you can run
-  `ig` in a Kubernetes pod and read the output directly.
-- If you are implementing an application that needs to get insights from the
-  Kubernetes node, you could include the `ig` binary in your container
+  `ig` in a Kubernetes pod and read the   output directly.
+- If you are implementing an application that needs to get insights from the Kubernetes node, you could include the `ig` binary 
+  in your container
   image, and your app simply execs it. In such a case, it is suggested to use
   the JSON output format to ease the parsing.
 - Outside a Kubernetes environment, for observing and debugging standalone
```
Comment (mauriciovasquezbernal) [inline docs/reference/ig.md:39]: I'm not sure about these changes. Please drop them, I'll try to work later on on that part.

File: docs/reference/ig.md [hunk 3]
```diff
@@ -107,63 +105,34 @@ RUNTIME.RUNTIMENAME RUNTIME.CONTAINERID
 docker              b72558e589cb95e835c4840de19f0306d4081091c34045246d62b6efed3549f4 myContainer
 ```
 
-### Common features
+### TLS Support
```
Comment (mauriciovasquezbernal) [inline docs/reference/ig.md:108]: btw, this should go under "Using ig as a daemon"

**Pull request closed without merging**